src/cryptonote_core/blockchain.h

```diff
@@ -1318,11 +1318,12 @@ namespace cryptonote
      *
      * get the block long term weights of the last <count> blocks, and return by reference <weights>.
      *
-     * @param weights return-by-reference the list of weights
      * @param start_height the block height of the first block to query
      * @param count the number of blocks to get weights for
+     *
+     * @return the long term median block weight
      */
-    void get_long_term_block_weights(std::vector<uint64_t>& weights, uint64_t start_height, size_t count) const;
+    uint64_t get_long_term_block_weight_median(uint64_t start_height, size_t count) const;
```

Comment:
I think the doxygen description also needs slight rewording (it just computes the median, instead of returning the weights):

>     * @brief gets recent block long term weights for median calculation
>     *
>     * get the block long term weights of the last <count> blocks, and return by reference <weights>.